pkg/kubelet/cm/memorymanager/state/state_checkpoint_test.go

```diff
@@ -34,20 +35,15 @@ const testingCheckpoint = "memorymanager_checkpoint_test"
 
 var testingDir = os.TempDir()
 
-// AssertStateEqual marks provided test as failed if provided states differ
-func AssertStateEqual(t *testing.T, restoredState, expectedState State) {
-	expectedMachineState := restoredState.GetMachineState()
+// assertStateEqual marks provided test as failed if provided states differ
+func assertStateEqual(t *testing.T, restoredState, expectedState State) {
+	expectedMachineState := expectedState.GetMachineState()
 	restoredMachineState := restoredState.GetMachineState()
-
-	if !reflect.DeepEqual(expectedMachineState, restoredMachineState) {
-		t.Errorf("Expected MachineState does not equal to restored one")
-	}
+	assert.Equal(t, expectedMachineState, restoredMachineState, "expected MachineState does not equal to restored one")
```

Comment:
Just curious: why `Equal` vs. `reflect.DeepEqual`

Reply:
I just started to use, `assert` library, IMHO it more unit testing style than all `if` conditions, and the `Equal` of the assert is doing the same as `reflect.DeepEqual`